### PR TITLE
PSP-5759 : Enhancement - Identify "Primary Contact" on acquisition de…

### PR DESCRIPTION
**Pull request by eddherrera**
image

image

### CODE REVIEW

File: source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.test.tsx [hunk 1]
```diff
@@ -37,6 +37,13 @@ describe('UpdateAcquisitionOwnersSubForm component', () => {
         utils.container.querySelector(
           `input[name="owners[${index}].otherName"]`,
         ) as HTMLInputElement,
+      getIsPrymaryContactRadioButtonValue: (index = 0) => {
```
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.test.tsx:40]: nit: Prymary -> Primary

File: source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx [hunk 2]
```diff
@@ -18,10 +18,40 @@ import {
 interface IUpdateAcquisitionOwnersSubFormProps {}
 
 const UpdateAcquisitionOwnersSubForm: React.FC<IUpdateAcquisitionOwnersSubFormProps> = () => {
-  const { values } = useFormikContext<WithAcquisitionOwners>();
+  const { values, setFieldValue, handleChange } = useFormikContext<WithAcquisitionOwners>();
   const [removeIndex, setRemoveIndex] = useState<number>(-1);
   const [showRemoveModal, setShowRemoveModal] = useState<boolean>(false);
 
+  const updatePrimaryContacts = (newPrimaryIndex: number) => {
```
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx:25]: These functions definitely need test coverage, but I don't see any tests in this PR, are those on the way?

**Review by devinleighsmith — APPROVED** [review]

**Comment by devinleighsmith** [comment]
@eddherrera other then test coverage, this one is looking good.

File: source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx (hunk 2 above)
Comment (eddherrera) [reply source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx:25]: Added tests to cover the scenarios for primary contact.

File: source/frontend/src/features/mapSideBar/tabs/SectionField.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { Col, Row } from 'react-bootstrap';
 import styled from 'styled-components';
 
 interface ISectionFieldProps {
-  label: string;
+  label: string | null;
```
Comment (FuriousLlama) [inline source/frontend/src/features/mapSideBar/tabs/SectionField.tsx:7]: Is it necessary to use the null? why not simply empty string?

File: source/frontend/src/features/properties/map/acquisition/common/models.ts [hunk 4]
```diff
@@ -76,6 +77,7 @@ export class AcquisitionOwnerFormModel {
       id: this.id,
       rowVersion: this.rowVersion,
       acquisitionFileId: this.acquisitionFileId,
+      isPrimaryContact: this.isPrimaryContact === 'true' ? true : false,
```
Comment (FuriousLlama) [inline source/frontend/src/features/properties/map/acquisition/common/models.ts:80]: although trivial, there are functions that perform the string 'true' to true conversion in our common libraries

File: source/frontend/src/features/mapSideBar/tabs/SectionField.tsx (hunk 3 above)
Comment (eddherrera) [reply source/frontend/src/features/mapSideBar/tabs/SectionField.tsx:7]: Is not necessary. But I would rather have the 'null' to convey the absence of a label.

File: source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx [hunk 5]
```diff
@@ -18,10 +18,40 @@ import {
...
+    if (values.owners.length > 1) {
+      for (let i = 0; i < values.owners.length; i++) {
+        if (i !== newPrimaryIndex) {
+          setFieldValue(`owners[${i}].isPrimaryContact`, '');
```
[the remark this answers is not in the document]
Comment (eddherrera) [reply source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx:29]: I would have used 'false' in the case where there's a radio button for such value.

File: source/frontend/src/features/properties/map/acquisition/common/models.ts (hunk 4 above)
Comment (eddherrera) [reply source/frontend/src/features/properties/map/acquisition/common/models.ts]: Updated

File: source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx [hunk 6]
```diff
@@ -18,10 +18,40 @@ import {
...
+  };
+
+  const onRemovedPrimaryContact = (index: number) => {
+    if (values.owners.length > 1) {
+      const isPrimary = values.owners[index].isPrimaryContact === 'true';
+      if (isPrimary) {
+        if (index === 0) {
+          values.owners[++index].isPrimaryContact = 'true';
+        } else {
+          values.owners[--index].isPrimaryContact = 'true';
```
[the remark this answers is not in the document]
Comment (eddherrera) [reply source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx]: Updated

File: source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (FuriousLlama) [reply source/frontend/src/features/properties/map/acquisition/common/update/acquisitionOwners/UpdateAcquisitionOwnersSubForm.tsx:29]: Ah got it. Either yes or no value

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**